backend/routes/db.py

```diff
@@ -343,16 +275,164 @@ async def get_concepts_post_route(request: Request, id: Union[List[str], None] =
     return await get_concepts_route(request, id=id, table=table)
 
 
+sessions_cache: Dict = {}
+session_stale_hours = 24
+
+
+@router.get("/start-session")
+def start_session():
+    return get_session(None)
+
+
+def get_session(session_id: int):
+    if session_id:
+        if session_id not in sessions_cache:
+            # raise HTTPException(status_code=404, detail="Session not found")
+            raise Exception(f"Session {session_id} not found")
+        return sessions_cache[session_id]
+    session = {
+        "session_id": next_api_call_group_id(),
+        "session_start": datetime.now(),
+        "concept_ids_seen": [],
+        "searches": {},
+    }
+    return session
+
+
+def update_session(session_id: int, concepts: List[Dict]):
+    session = get_session(session_id)
+    session["concept_ids"].extend([c["concept_id"] for c in concepts])
+    session["last_used"] = datetime.now()
+    sessions_cache[session_id] = session
+    clear_old_sessions()
+    return session
+
+
+@throttle(wait = 600) # 10 minutes
```

Comment:
Oh interesting, what does this do? I've never seen.

Does this come from this package?: https://pypi.org/project/throttle/
If so, we should:
- [ ] Update `requirements.txt` and `requirements-unlocked.txt`

_edit_: I see now you made your own throttle decorator. If you're having any issues with it, or if we just prefer to use an external package, we could do so, but I'm fine with either.

Reply:
So the idea is that we limit session clearing to 1x every 10 minutes, and when we trigger a clear, we clear anything older than 24 hours?

Reply:
I assume the throttling is to save on performance, but does it help that much? Is this function really going to get triggered that often and, if so, is it really that taxing to clear attempt to clear the cache?